sql/core/src/main/scala/org/apache/spark/sql/execution/datasources/csv/CSVOptions.scala

```diff
@@ -101,3 +125,14 @@ private[sql] class CSVOptions(
 
   val rowSeparator = "\n"
 }
+
+object CSVOptions {
+
+  /** Used for convenient construction in unit tests */
+  def apply(): CSVOptions = new CSVOptions(Map.empty)
```

Comment:
For me, I a bit hesitated if this `CSVOptions` companion object is only used in unit tests.

I'd just use `new CSVOptions(Map("key" -> "value"))` or `new CSVOptions(Map.empty)` in tests.
Otherwise, I'd just make this object in the tests if this object is required for some reasons or just make a function in tests for convenient construction.